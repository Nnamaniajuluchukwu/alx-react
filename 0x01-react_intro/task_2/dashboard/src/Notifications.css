.Notifications {
    border: 3px dashed #e0354b;
    padding: 1rem;
}

li[data-priority='default'] {
    color: blue;
}

li[data-priority='urgent'], li {
    color: red;
}
