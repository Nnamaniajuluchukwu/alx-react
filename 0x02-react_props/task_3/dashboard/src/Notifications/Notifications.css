.Notifications {
    border: 3px dashed #e0354b;
    padding: 1rem;
}

li[data-notification-type='default'] {
    color: blue;
}

li[data-notification-type='urgent'], li {
    color: red;
}
