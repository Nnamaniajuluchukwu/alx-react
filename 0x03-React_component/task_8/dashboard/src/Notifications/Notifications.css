.Notifications {
    border: 3px dashed #e0354b;
    padding: 1rem;
    width: 30%;
    position: absolute;
    right: 0;
}

.menuItem {
    text-align: right;
}

li[data-notification-type='default'] {
    color: blue;
}

li[data-notification-type='urgent'], li {
    color: red;
}
