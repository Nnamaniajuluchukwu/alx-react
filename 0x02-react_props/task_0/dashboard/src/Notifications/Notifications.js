import React from "react";
import './Notifications.css';
import { getLatestNotification } from "../utils/utils";


export default function Notifications() {
    return (
        <div className="Notifications">
            <p>Here is the list of notifications</p>
            <ul>
                <li data-priority="default">New course available</li>
                <li data-priority="urgent">New resume available</li>
                <li dangerouslySetInnerHTML={{ __html: getLatestNotification() }}></li>
            </ul>
            <button style={{
                float: 'right',
                marginTop: '-9.5em',
                marginRight: '-0.5em',
                backgroundColor: '#fff',
            }} aria-label='Close' onClick={() => console.log('Close button has been clicked')}>
                <img src={require('../assets/close-button.png')} height="15px" width="13px" alt="close icon"/>
            </button>
        </div>
    );
}
