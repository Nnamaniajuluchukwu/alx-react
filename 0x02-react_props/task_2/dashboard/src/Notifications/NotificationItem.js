import React from "react";
import './Notifications.css';


export default function NotificationItem({type, html , value}) {
    if (value) return <li data-notification-type={type}> {value} </li>
    else return <li dangerouslySetInnerHTML={html}></li>
}
