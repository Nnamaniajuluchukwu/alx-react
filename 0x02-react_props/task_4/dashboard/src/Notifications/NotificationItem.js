import React from "react";
import { PropTypes } from 'prop-types';
import './Notifications.css';


export default function NotificationItem({type='default', html , value}) {
    if (value) return <li data-notification-type={type}> {value} </li>
    else return <li dangerouslySetInnerHTML={html}></li>
}

NotificationItem.propTypes = {
    type: PropTypes.string,
    value: PropTypes.string,
    html: PropTypes.shape({ 
      __html: PropTypes.string
    }),
};
