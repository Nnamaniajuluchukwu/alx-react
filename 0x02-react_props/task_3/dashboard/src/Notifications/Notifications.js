import React from "react";
import './Notifications.css';
import { getLatestNotification } from "../utils/utils";
import NotificationItem from "./NotificationItem";


export default function Notifications() {
    return (
        <div className='Notifications'>
            <p>Here is the list of notifications</p>
            <ul>
                <NotificationItem type='default' value='New course available' />
                <NotificationItem type='urgent' value='New resume available' />
                <NotificationItem html={{ __html: getLatestNotification() }} />
            </ul>
            <button style={{
                float: 'right',
                marginTop: '-9.5em',
                marginRight: '-0.5em',
                backgroundColor: '#fff',
            }} aria-label='Close' onClick={() => console.log('Close button has been clicked')}>
                <img src={require('../assets/close-button.png')} height="15px" width="13px" alt="close icon"/>
            </button>
        </div>
    );
}
