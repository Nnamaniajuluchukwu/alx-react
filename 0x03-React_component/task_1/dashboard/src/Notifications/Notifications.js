import React from "react";
import './Notifications.css';
import PropTypes from 'prop-types';
// import { getLatestNotification } from "../utils/utils";
import NotificationItem from "./NotificationItem";
import NotificationItemshape from "./NotificationItemShape";


export default function Notifications({ displayDrawer=false, listNotification }) {
    let notifyMessage;

    if (listNotification.length === 0) {
        notifyMessage = 'No new notification for now';
    } else {
        notifyMessage = listNotification.map((notify) => {
            return <NotificationItem key={notify.id} type={notify.type} value={notify.value} html={notify.html} />
        })
    }

    return (
        <>
            <div className="menuItem">
                <p>Your notifications</p>
            </div>
            {displayDrawer ? 
                <div className='Notifications'>
                    <p>Here is the list of notifications</p>
                    <ul>
                        {notifyMessage}
                    </ul>
                    <button style={{
                        float: 'right',
                        marginTop: '-7.0em',
                        marginRight: '-0.5em',
                        backgroundColor: '#fff',
                    }} aria-label='Close' onClick={() => console.log('Close button has been clicked')}>
                        <img src={require('../assets/close-button.png')} height="15px" width="12px" alt="close icon" />
                    </button>
                </div>
            : ''}
        </>
    );
}

Notifications.propTypes = {
    displayDrawer: PropTypes.bool,
    listNotification: PropTypes.arrayOf(NotificationItemshape)
}
