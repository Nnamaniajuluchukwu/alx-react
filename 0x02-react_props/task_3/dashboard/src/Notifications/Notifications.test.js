import React from 'react';
import { shallow } from 'enzyme';
import Notifications from './Notifications';
import toJson from 'enzyme-to-json';
import NotificationItem from './NotificationItem';


describe('Notifications Component', () => {
    let wrapper;

    beforeAll(() => {
        wrapper = shallow(<Notifications />);
    })

    it('should render without crashing', () => {
        expect(wrapper.exists()).toBeTruthy();
    });

    it('renders three list items', () => {
        const listItems = wrapper.find(NotificationItem);
        expect(listItems.length).toEqual(3);
    });

    it("renders the text 'Here is the list of notifications'", () => {
        const text = wrapper.find('p');
        expect(text.props().children).toEqual('Here is the list of notifications');
    });

    it('should match the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
