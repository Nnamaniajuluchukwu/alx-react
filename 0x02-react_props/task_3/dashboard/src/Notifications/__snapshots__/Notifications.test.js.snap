// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Component should match the snapshot 1`] = `
<div
  className="Notifications"
>
  <p>
    Here is the list of notifications
  </p>
  <ul>
    <NotificationItem
      type="default"
      value="New course available"
    />
    <NotificationItem
      type="urgent"
      value="New resume available"
    />
    <NotificationItem
      html={
        Object {
          "__html": "<strong>Urgent requirement</strong> - complete by EOD",
        }
      }
    />
  </ul>
  <button
    aria-label="Close"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "float": "right",
        "marginRight": "-0.5em",
        "marginTop": "-9.5em",
      }
    }
  >
    <img
      alt="close icon"
      height="15px"
      src="mock-file"
      width="13px"
    />
  </button>
</div>
`;
